File: 10773/10773.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   10773.cpp                                          :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/02/18 14:29:02 by jihoolee          #+#    #+#             */
/*   Updated: 2024/02/18 14:36:03 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <stack>
#include <numeric>

int get_sum(std::stack<int>* p_s) {
  int sum = 0;

  while (!p_s->empty()) {
    sum += p_s->top();
    p_s->pop();
  }
  return sum;
}

int main(void) {
  std::stack<int> s;
  int             num_cnt;
  int             num;

  std::cin >> num_cnt;
  for (int i = 0; i < num_cnt; ++i) {
    std::cin >> num;
    if (num == 0)
      s.pop();
    else
      s.push(num);
  }
  std::cout << get_sum(&s);
  return 0;
}

File: 10773/10773.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    10773.py                                           :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2024/02/18 14:36:55 by jihoolee          #+#    #+#              #
#    Updated: 2024/02/18 14:40:45 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

import sys


input = sys.stdin.readline


def main() -> None:
    l: list[int] = []

    for _ in range(int(input())):
        num = int(input())
        if (num) == 0:
            l.pop()
        else:
            l.append(num)
    print(sum(l))


if __name__ == '__main__':
    main()
